train_dreambooth.py

```diff
@@ -148,32 +166,44 @@ def run(args):
     train_dataset = data_util.prepare_datasets()
 
     print("Initializing trainer...")
-    ckpt_path_prefix = (
-        run_name
-    ) = f"lr@{args.lr}-max_train_steps@{args.max_train_steps}-train_text_encoder@{args.train_text_encoder}"
+    ckpt_path_prefix = run_name
     dreambooth_trainer = prepare_trainer(
         args.img_resolution, args.train_text_encoder, args.mp
     )
-    train(dreambooth_trainer, train_dataset, ckpt_path_prefix, args.max_train_steps)
 
-    if args.log_wandb:
-        print("Logging artifacts...")
-        wandb.init(project="dreambooth-keras", name=run_name, config=vars(args))
-        if args.validation_prompt is None:
-            args.validation_prompt = (
-                f"A photo of {args.unique_id} {args.class_category} in a bucket"
+    callbacks = (
+        [
+            WandbMetricsLogger(log_freq="batch"),
+            DreamBoothCheckpointCallback(
+                ckpt_path_prefix, save_weights_only=True, monitor="loss", mode="min"
+            ),
+            QualitativeValidationCallback(
+                img_heigth=args.img_resolution,
+                img_width=args.img_resolution,
+                prompts=validation_prompts,
+                num_imgs_to_gen=args.num_images_to_generate,
+            ),
+        ]
+        if args.log_wandb
+        else [
+            tf.keras.callbacks.ModelCheckpoint(
+                ckpt_path_prefix, save_weights_only=True, monitor="loss", mode="min"
             )
+        ]
+    )
```

Comment:
The problem with only saving the low-loss checkpoints in DreamBooth is that the loss curves are not that informative if you take a look at them (think of vanilla GAN training loss curves).

Reply:
So, this is why I just logged weights after the end of training. 

In my experience, the low-loss checkpoints often lead to inferior results.

Reply:
I'll remove the monitor in that case.